plugins/pulp_docker/plugins/importers/importer.py

```diff
@@ -392,16 +440,86 @@ def _remove_manifest(cls, repo, manifest):
         cls._purge_unlinked_tags(repo, manifest)
         cls._purge_unlinked_blobs(repo, manifest)
 
+    @classmethod
+    def _remove_manifest_list(cls, repo, manifest_list):
+        """
+        Purge Tags and image manifests associated with a given ManifestList in the repository.
+
+        :param repo:     The affected repository.
+        :type  repo:     pulp.server.db.model.Repository
+        :param manifest_list: The ManifestList being removed
+        :type  manifest_list: pulp_docker.plugins.models.ManifestList
+        """
+        cls._purge_unlinked_tags(repo, manifest_list)
+        cls._purge_unlinked_manifests(repo, manifest_list)
+
+    @staticmethod
+    def _purge_unlinked_manifests(repo, manifest_list):
+
+        # Find manifest digests referenced by removed manifest lists (orphaned)
+        orphaned = set()
+        for image_man in manifest_list.manifests:
+            orphaned.add(image_man)
+            if manifest_list.amd64_digest:
+                orphaned.add(manifest_list.amd64_digest)
+        if not orphaned:
+            # nothing orphaned
+            return
+
+        # Find manifest digests still referenced by other manifest lists (adopted)
+        adopted = set()
+        criteria = UnitAssociationCriteria(type_ids=[constants.MANIFEST_LIST_TYPE_ID],
+                                           unit_filters={'digest': {'$ne': manifest_list.digest}})
+        for man_list in unit_association.RepoUnitAssociationManager._units_from_criteria(
+                repo, criteria):
+            for image_man in man_list.manifests:
+                adopted.add(image_man)
+            if man_list.amd64_digest:
+                adopted.add(man_list.amd64_digest) 
+
+        # Remove unreferenced manifests
+        orphaned = orphaned.difference(adopted)
```

Comment:
FWIW, I would find this a bit more readable and be more confident that I know what it means if it were written as ``orphaned = orphaned - adopted``, which you could even shorten to ``orphaned -= adopted``. That may just be a matter of preference though. Go with whatever you like.